pkg/action/executor/resize_container_util.go

```diff
@@ -97,32 +97,36 @@ func checkLimitsRequests(container *k8sapi.Container) error {
 	return nil
 }
 
-func updateResourceAmount(podSpec *k8sapi.PodSpec, spec *containerResizeSpec) (bool, error) {
-	//1. get container
-	index := spec.Index
-	if index >= len(podSpec.Containers) {
-		return false, fmt.Errorf("failed to find container[%d] in pod", index)
-	}
-	container := &(podSpec.Containers[index])
-
-	//2. update Limits
+func updateResourceAmount(podSpec *k8sapi.PodSpec, specs []*containerResizeSpec) (bool, error) {
 	changed := false
-	if spec.NewCapacity != nil && len(spec.NewCapacity) > 0 {
-		changed = changed || updateLimits(container, spec.NewCapacity)
-	}
+	for _, spec := range specs {
+		//1. get container
+		index := spec.Index
+		if index >= len(podSpec.Containers) {
+			// TODO: we can as well try updating spec for other containers ignoring this one.
+			return false, fmt.Errorf("failed to find container[%d] in pod", index)
+		}
+		container := &(podSpec.Containers[index])
 
-	//3. update Requests
-	if spec.NewRequest != nil && len(spec.NewRequest) > 0 {
-		changed = changed || updateRequests(container, spec.NewRequest)
-	}
+		//2. update Limits
+		if spec.NewCapacity != nil && len(spec.NewCapacity) > 0 {
+			changed = updateLimits(container, spec.NewCapacity)
```

Comment:
If the global `changed` bool is updated in such way, the final returned `changed` bool (in line 132) only represents whether request is changed for the last container in the `specs` slice.

Maybe we can introduce a temporary bool at the start of each iteration (under line 102) like
```
containerResChanged := false
```
and update this local bool here:
```
containerResChanged = containerResChanged || updateLimits(container, spec.NewCapacity)
```
and same at line 118.

In line 127, we can just check `if !containerResChanged`.

Furthermore, at the end of each iteration, update the global `changed` bool
```
changed = changed || containerResChanged
```

Reply:
Nice catch @marvelyue and thanks for having a look. Fixed!
Also apologies for a late reply. I somehow missed the notification for this comment and found it today when I came here to check as to what's missing for this PR to get merged.